### ISSUE TITLE
[url_launcher] flutter web not working on iPhone safari when use http request

### ISSUE THREAD

**Issue by figbin** [flutter/flutter#113172]
```
import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;
import 'package:url_launcher/url_launcher.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {


  @override
  void initState() {
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Container(
        color: Colors.red,
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: () async {
          try{
            await http.get(Uri.parse('https://www.google.com/')); //any url support access-control-allow-origin *
          }catch(e){

          }
          //remove above http request can be open
          launchUrl(Uri.parse('https://www.google.com/'), mode: LaunchMode.externalApplication, webOnlyWindowName: '_blank');
          // above can't open new url on iPhone safari, but PC was OK.
        },
        tooltip: 'TestClick',
        child: const Icon(Icons.add),
      ),
    );
  }
}
```

**Comment by figbin** [comment]
```
flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, 3.4.0-35.0.pre.2, on macOS 12.0.1 21A559 darwin-x64, locale zh-Hans-CN)
[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0-rc5)
[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.2)
[✓] VS Code (version 1.71.2)
[✓] Connected device (2 available)
```

**Comment by danagbemava-nc** [comment]
Hi @figbin, I don’t think this is a flutter issue but rather a feature of safari. I tested your code sample and I noticed that safari was blocking popups on the desktop version as well as the mobile web version when the HTTP request failed. After disabling `block popups` in safari settings on the iOS simulator, I get a popup that makes it so I decide whether or not the popup is blocked. In this case, when I allow navigation, the page loads as expected. 
See the recording below. 

<details>
<summary>video</summary>

https://user-images.githubusercontent.com/88313112/194845651-14a7f440-7ccf-4d33-8645-af81f74dd609.mp4
</details>
